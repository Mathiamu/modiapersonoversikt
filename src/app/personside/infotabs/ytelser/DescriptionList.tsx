import * as React from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { EtikettLiten, Normaltekst } from 'nav-frontend-typografi';

interface Props {
    entries: DescriptionListEntries;
}

type DescriptionlistEntry = string | number | null;

export interface DescriptionListEntries {
    [name: string]: DescriptionlistEntry;
}

function getDescriptionlistEntry(term: string, description: DescriptionlistEntry) {
    return (
        <div key={term}>
            <EtikettLiten tag="dt">{term || ''}</EtikettLiten>
            <Normaltekst tag="dd">{description || ''}</Normaltekst>
        </div>
    );
}

function createDescriptionListEntries(valuePairs: DescriptionListEntries) {
    return Object.keys(valuePairs).map(key =>
        getDescriptionlistEntry(key, valuePairs[key])
    );
}

const ListStyling = styled.dl`
    display: flex;
    flex-wrap: wrap;
    dt {
      color: ${theme.color.grÃ¥Skrift};
    }
    dd {
        font-weight: bold;
        margin-top: .3rem;
    }
    > div {
        margin-top: ${theme.margin.px30};
        padding-right: 1rem;
        min-width: 13rem;
    }
`;

function DescriptionList(props: Props) {
    return (
        <ListStyling>
            {createDescriptionListEntries(props.entries)}
        </ListStyling>
    );
}

export default DescriptionList;
