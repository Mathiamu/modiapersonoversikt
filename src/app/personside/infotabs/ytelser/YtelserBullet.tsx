import * as React from 'react';
import { ReactNode } from 'react';
import theme from '../../../../styles/personOversiktTheme';
import { BulletPoint } from '../../../../components/common-styled-components';
import { Ingress } from 'nav-frontend-typografi';
import styled from 'styled-components';

interface Props {
    children: ReactNode;
    tittel: string;
}

const Wrapper = styled.div`
    padding: ${theme.margin.px30} 0;
`;

function YtelserBullet(props: Props) {
    return (
        <Wrapper>
            <BulletPoint showBulletPoint={true} color={theme.color.ytelser}>
                <Ingress>{props.tittel}</Ingress>
            </BulletPoint>
            {props.children}
        </Wrapper>
    );
}

export default YtelserBullet;
