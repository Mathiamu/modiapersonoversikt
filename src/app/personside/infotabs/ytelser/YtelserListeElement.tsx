import * as React from 'react';
import {
    getYtelseIdDato,
    isForeldrepenger,
    isPleiepenger,
    isSykepenger,
    Ytelse
} from '../../../../models/ytelse/ytelse-utils';
import VisMerKnapp from '../../../../components/VisMerKnapp';
import { getUnikPleiepengerKey } from '../../../../models/ytelse/pleiepenger';
import { Element, Normaltekst, Undertittel } from 'nav-frontend-typografi';
import { formaterDato } from '../../../../utils/stringFormatting';
import { getUnikSykepengerKey } from '../../../../models/ytelse/sykepenger';
import { getForeldepengerIdDato, getUnikForeldrepengerKey } from '../../../../models/ytelse/foreldrepenger';
import { useInfotabsDyplenker } from '../dyplenker';

interface Props {
    ytelse: Ytelse;
}

function YtelserListeElement(props: Props) {
    const dypLenker = useInfotabsDyplenker();

    if (isPleiepenger(props.ytelse)) {
        return (
            <li>
                <VisMerKnapp
                    key={getUnikPleiepengerKey(props.ytelse)}
                    ariaDescription="Vis pleiepenger"
                    valgt={dypLenker.ytelser.erValgt(props.ytelse)}
                    linkTo={dypLenker.ytelser.link(props.ytelse)}
                >
                    <Undertittel tag={'h3'}>Pleiepenger sykt barn</Undertittel>
                    <Element>ID-dato</Element>
                    <Normaltekst>{formaterDato(getYtelseIdDato(props.ytelse))}</Normaltekst>
                    <Element>Barnets f.nr: </Element>
                    <Normaltekst>{props.ytelse.barnet}</Normaltekst>
                </VisMerKnapp>
            </li>
        );
    }

    if (isSykepenger(props.ytelse)) {
        return (
            <li>
                <VisMerKnapp
                    key={getUnikSykepengerKey(props.ytelse)}
                    ariaDescription="Vis sykepenger"
                    valgt={dypLenker.ytelser.erValgt(props.ytelse)}
                    linkTo={dypLenker.ytelser.link(props.ytelse)}
                >
                    <Undertittel tag={'h3'}>Sykepengerrettighet</Undertittel>
                    <Element>ID-dato</Element>
                    <Normaltekst>{formaterDato(getYtelseIdDato(props.ytelse))}</Normaltekst>
                </VisMerKnapp>
            </li>
        );
    }

    if (isForeldrepenger(props.ytelse)) {
        return (
            <li>
                <VisMerKnapp
                    key={getUnikForeldrepengerKey(props.ytelse)}
                    ariaDescription="Vis foreldrepenger"
                    valgt={dypLenker.ytelser.erValgt(props.ytelse)}
                    linkTo={dypLenker.ytelser.link(props.ytelse)}
                >
                    <Undertittel tag={'h3'}>Foreldrepenger</Undertittel>
                    <Element>ID-dato</Element>
                    <Normaltekst>{formaterDato(getForeldepengerIdDato(props.ytelse))}</Normaltekst>
                </VisMerKnapp>
            </li>
        );
    }

    return null;
}

export default YtelserListeElement;
