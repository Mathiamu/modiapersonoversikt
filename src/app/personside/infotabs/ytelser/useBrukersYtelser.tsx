import { useOnMount, useRestResource } from '../../../../utils/customHooks';
import { useDispatch } from 'react-redux';
import { hasData, isLoading, isNotStarted } from '../../../../rest/utils/restResource';
import { ReactNode, useMemo } from 'react';
import { getForeldepengerIdDato } from './foreldrepenger/ForeldrepengerEkspanderbartPanel';
import { getPleiepengerIdDato } from './pleiepenger/PleiepengerEkspanderbartPanel';
import { getSykepengerIdDatp } from './sykepenger/SykepengerEkspanderbartPanel';
import { datoSynkende } from '../../../../utils/dateUtils';
import { Pleiepengerettighet } from '../../../../models/ytelse/pleiepenger';
import { Sykepenger } from '../../../../models/ytelse/sykepenger';
import { Foreldrepengerettighet } from '../../../../models/ytelse/foreldrepenger';

interface YtelseMedDato {
    idDato: string;
    markup: ReactNode;
}

interface Props {
    pleiepengerKomponent: (pleiepenger: Pleiepengerettighet, key: string) => ReactNode;
    sykepengerKomponent: (sykepenger: Sykepenger, key: string) => ReactNode;
    foreldrepengerKomponent: (foreldrepenger: Foreldrepengerettighet, key: string) => ReactNode;
}

function useBrukersYtelser(props: Props) {
    const foreldrepengerResource = useRestResource(resources => resources.foreldrepenger);
    const pleiepengerResource = useRestResource(resources => resources.pleiepenger);
    const sykepengerResource = useRestResource(resources => resources.sykepenger);
    const dispatch = useDispatch();

    useOnMount(() => {
        if (isNotStarted(foreldrepengerResource)) {
            dispatch(foreldrepengerResource.actions.fetch);
        }
        if (isNotStarted(pleiepengerResource)) {
            dispatch(pleiepengerResource.actions.fetch);
        }
        if (isNotStarted(sykepengerResource)) {
            dispatch(sykepengerResource.actions.fetch);
        }
    });

    const foreldrepenger: YtelseMedDato[] = useMemo(
        () =>
            hasData(foreldrepengerResource) && foreldrepengerResource.data.foreldrepenger
                ? foreldrepengerResource.data.foreldrepenger.map(foreldrepengerettighet => {
                      const idDato = getForeldepengerIdDato(foreldrepengerettighet);
                      return {
                          idDato: idDato,
                          markup: props.foreldrepengerKomponent(foreldrepengerettighet, 'foreldrepenger' + idDato)
                      };
                  })
                : [],
        [foreldrepengerResource, props]
    );

    const pleiepenger: YtelseMedDato[] = useMemo(
        () =>
            hasData(pleiepengerResource) && pleiepengerResource.data.pleiepenger
                ? pleiepengerResource.data.pleiepenger.map(pleiepengerettighet => {
                      const idDato = getPleiepengerIdDato(pleiepengerettighet);
                      return {
                          idDato: idDato,
                          markup: props.pleiepengerKomponent(pleiepengerettighet, 'pleiepenger' + idDato)
                      };
                  })
                : [],
        [pleiepengerResource, props]
    );

    const sykepenger: YtelseMedDato[] = useMemo(
        () =>
            hasData(sykepengerResource) && sykepengerResource.data.sykepenger
                ? sykepengerResource.data.sykepenger.map(sykepengerettighet => {
                      const idDato = getSykepengerIdDatp(sykepengerettighet);
                      return {
                          idDato: idDato,
                          markup: props.sykepengerKomponent(sykepengerettighet, 'sykepenger' + idDato)
                      };
                  })
                : [],
        [sykepengerResource]
    );

    const rettigheter = useMemo(
        () =>
            [...foreldrepenger, ...pleiepenger, ...sykepenger]
                .sort(datoSynkende(rettighet => rettighet.idDato))
                .map(rettighet => rettighet.markup),
        [foreldrepenger, pleiepenger, sykepenger]
    );

    const pending =
        isLoading(pleiepengerResource) || isLoading(foreldrepengerResource) || isLoading(sykepengerResource);

    return { rettigheter, pending };
}

export default useBrukersYtelser;
