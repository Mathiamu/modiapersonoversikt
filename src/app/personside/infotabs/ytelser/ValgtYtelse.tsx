import * as React from 'react';
import { useRef } from 'react';
import styled from 'styled-components/macro';
import VisuallyHiddenAutoFokusHeader from '../../../../components/VisuallyHiddenAutoFokusHeader';
import theme from '../../../../styles/personOversiktTheme';
import { erModiabrukerdialog } from '../../../../utils/erNyPersonoversikt';
import { isForeldrepenger, isPleiepenger, isSykepenger, Ytelse } from '../../../../models/ytelse/ytelse-utils';
import Foreldrepenger from './foreldrepenger/ForeldrePenger';
import Pleiepenger from './pleiepenger/Pleiepenger';
import Sykepenger from './sykepenger/Sykepenger';
import { loggError } from '../../../../utils/frontendLogger';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { guid } from 'nav-frontend-js-utils';
import { useInfotabsDyplenker } from '../dyplenker';
import useBrukersYtelser from './useBrukersYtelser';
import { usePrevious } from '../../../../utils/customHooks';
import { useEffect } from 'react';

interface Props {
    valgtYtelse?: Ytelse;
}

const Styling = styled.section`
    > * {
        margin-bottom: 0.5rem;
    }
    padding: ${theme.margin.layout};
`;

function YtelseMarkup(props: { ytelse: Ytelse }) {
    if (isForeldrepenger(props.ytelse)) {
        return <Foreldrepenger foreldrepenger={props.ytelse} />;
    }
    if (isPleiepenger(props.ytelse)) {
        return <Pleiepenger pleiepenger={props.ytelse} />;
    }
    if (isSykepenger(props.ytelse)) {
        return <Sykepenger sykepenger={props.ytelse} />;
    }
    loggError(new Error('Ytelse ikke h√•ndtert, kunne ikke finne markup'));
    return null;
}

function ValgtYtelse(props: Props) {
    const ytelser = useBrukersYtelser();
    const tittelRef = useRef<HTMLHeadingElement>(null);
    const dypLenker = useInfotabsDyplenker();
    const valgtYtelse = ytelser.ytelser.find(ytelse => dypLenker.ytelser.erValgt(ytelse));
    const prevYtelse = usePrevious(valgtYtelse);
    useEffect(
        function focusVedNyYtelse() {
            if (!valgtYtelse || !prevYtelse) {
                return;
            }
            if (prevYtelse !== valgtYtelse) {
                tittelRef.current && tittelRef.current.focus();
            }
        },
        [valgtYtelse, prevYtelse, tittelRef]
    );

    const titleId = useRef(guid());
    if (!props.valgtYtelse) {
        return <AlertStripeInfo>Fant ingen ytelser for bruker</AlertStripeInfo>;
    }

    return (
        <Styling>
            <h2 className="sr-only" id={titleId.current} ref={tittelRef} tabIndex={-1}>
                Valgt ytelse
            </h2>
            {erModiabrukerdialog() && <VisuallyHiddenAutoFokusHeader tittel="Ytelser" />}
            <YtelseMarkup ytelse={props.valgtYtelse} />
        </Styling>
    );
}

export default ValgtYtelse;
