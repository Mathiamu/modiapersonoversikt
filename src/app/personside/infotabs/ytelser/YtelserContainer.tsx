import * as React from 'react';
import styled from 'styled-components';
import VisuallyHiddenAutoFokusHeader from '../../../../components/VisuallyHiddenAutoFokusHeader';
import theme from '../../../../styles/personOversiktTheme';
import { erModiabrukerdialog } from '../../../../utils/erNyPersonoversikt';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import useBrukersYtelser from './useBrukersYtelser';

const Styling = styled.section`
    > * {
        margin-bottom: 0.5rem;
    }
    ${theme.resetEkspanderbartPanelStyling};
`;

function YtelserContainer() {
    const { rettigheter, pending } = useBrukersYtelser();

    return (
        <Styling>
            {erModiabrukerdialog() && <VisuallyHiddenAutoFokusHeader tittel="Ytelser" />}
            {rettigheter}
            {!pending && rettigheter.length === 0 && <AlertStripeInfo>Ingen ytelser funnet for bruker</AlertStripeInfo>}
            {pending && <CenteredLazySpinner />}
        </Styling>
    );
}

export default YtelserContainer;
