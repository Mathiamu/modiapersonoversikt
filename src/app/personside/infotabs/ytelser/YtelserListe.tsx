import useBrukersYtelser from './useBrukersYtelser';
import { getPleiepengerIdDato, getUnikPleiepengerKey } from '../../../../models/ytelse/pleiepenger';
import { getUnikSykepengerKey } from '../../../../models/ytelse/sykepenger';
import { getUnikForeldrepengerKey } from '../../../../models/ytelse/foreldrepenger';
import * as React from 'react';
import VisMerKnapp from '../../../../components/VisMerKnapp';
import { formaterDato } from '../../../../utils/stringFormatting';
import { Element, Normaltekst, Undertittel } from 'nav-frontend-typografi';
import VisuallyHiddenAutoFokusHeader from '../../../../components/VisuallyHiddenAutoFokusHeader';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { erModiabrukerdialog } from '../../../../utils/erNyPersonoversikt';

const Styling = styled.section`
    ${theme.hvittPanel};
`;

const YtelserListeStyle = styled.ol`
    > *:not(:first-child) {
        border-top: ${theme.border.skille};
    }
`;

function YtelseListe() {
    const { ytelser, pending, feilmeldinger } = useBrukersYtelser({
        renderPleiepenger: pleiepenger => (
            <VisMerKnapp key={getUnikPleiepengerKey(pleiepenger)} ariaDescription="Vis pleiepenger" valgt={false}>
                <Undertittel>Pleiepenger sykt barn</Undertittel>
                <Element>ID-dato</Element>
                <Normaltekst>{formaterDato(getPleiepengerIdDato(pleiepenger))}</Normaltekst>
                <Element>Barnets f.nr:</Element>
                <Normaltekst>{pleiepenger.barnet}</Normaltekst>
            </VisMerKnapp>
        ),
        renderSykepenger: sykepenger => (
            <VisMerKnapp key={getUnikSykepengerKey(sykepenger)} ariaDescription="Vis sykepenger" valgt={false}>
                <Undertittel>Sykepengerrettighet</Undertittel>
                <Element>ID-dato</Element>
                <Normaltekst>{formaterDato(getUnikSykepengerKey(sykepenger))}</Normaltekst>
                <Element>f.nr:</Element>
                <Normaltekst>{sykepenger.fødselsnummer}</Normaltekst>
            </VisMerKnapp>
        ),
        renderForeldrepenger: foreldrepenger => (
            <VisMerKnapp
                key={getUnikForeldrepengerKey(foreldrepenger)}
                ariaDescription="Vis foreldrepenger"
                valgt={false}
            >
                <Undertittel>Pleiepenger sykt barn</Undertittel>
                <Element>ID-dato</Element>
                <Normaltekst>{formaterDato(getUnikForeldrepengerKey(foreldrepenger))}</Normaltekst>
                <Element>Barnets f.nr:</Element>
                <Normaltekst>{foreldrepenger.barnetsFødselsdato}</Normaltekst>
            </VisMerKnapp>
        )
    });

    return (
        <Styling>
            {erModiabrukerdialog() && <VisuallyHiddenAutoFokusHeader tittel="Ytelser" />}
            {feilmeldinger}
            {!pending && feilmeldinger.length === 0 && ytelser.length === 0 && (
                <AlertStripeInfo>Ingen ytelser funnet for bruker</AlertStripeInfo>
            )}
            {pending && <CenteredLazySpinner />}
            <YtelserListeStyle role="tablist">{ytelser}</YtelserListeStyle>
        </Styling>
    );
}

export default YtelseListe;
