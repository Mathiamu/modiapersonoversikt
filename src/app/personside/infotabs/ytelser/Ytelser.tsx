import * as React from 'react';

import YtelseListe from './YtelserListe';
import { ScrollBar } from '../utils/InfoTabsScrollBar';
import styled from 'styled-components';
import ValgtYtelse from './ValgtYtelse';

const Styling = styled.section`
    flex-grow: 1; /* IE11 */
    display: flex;
    > *:first-child {
        min-width: 19rem;
        flex-basis: 19rem;
        flex-grow: 0;
        padding-top: 1rem;
    }
    > *:last-child {
        flex-grow: 1;
    }
    align-items: flex-start;
`;

function Ytelser() {
    return (
        <Styling>
            <ScrollBar keepScrollId="ytelser">
                <YtelseListe />
            </ScrollBar>
            <ScrollBar keepScrollId="ytelser">
                <ValgtYtelse />
            </ScrollBar>
        </Styling>
    );
}

export default Ytelser;
