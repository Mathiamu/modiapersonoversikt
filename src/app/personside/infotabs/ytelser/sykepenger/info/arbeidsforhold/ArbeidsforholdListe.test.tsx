import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ArbeidsForholdListe from './ArbeidsforholdListe';
import { statiskSykepengerMock } from '../../../../../../../mock/ytelse/statiskSykepengerMock';
import TestProvider from '../../../../../../../test/Testprovider';
import { getTestStore } from '../../../../../../../setupTests';
import { toggleVisAlleArbeidsforholdSykepengerActionCreator } from '../../../../../../../redux/ytelser/sykepengerReducer';

test('ArbeidsforholdListe matcher snapshot', () => {
    const arbeidsforhold = statiskSykepengerMock.arbeidsforholdListe[0];
    const resultat = renderer.create(
        <TestProvider>
            <ArbeidsForholdListe arbeidsforhold={[arbeidsforhold]} />
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});

test('ArbeidsforholdListe med alle vis alle arbeidsforhold matcher snapshot', () => {
    const arbeidsforhold = [statiskSykepengerMock.arbeidsforholdListe[0], statiskSykepengerMock.arbeidsforholdListe[0]];

    const testStore = getTestStore();
    testStore.dispatch(toggleVisAlleArbeidsforholdSykepengerActionCreator());
    const resultat = renderer.create(
        <TestProvider customStore={testStore}>
            <ArbeidsForholdListe arbeidsforhold={arbeidsforhold} />
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});
