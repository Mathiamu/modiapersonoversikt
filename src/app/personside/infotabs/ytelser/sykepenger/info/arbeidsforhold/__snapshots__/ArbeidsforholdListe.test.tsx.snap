// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArbeidsforholdListe matcher snapshot 1`] = `
.c1 span:not(:last-child):after {
  content: '.';
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 dt {
  color: #78706a;
}

.c0 dd {
  font-weight: bold;
  margin-top: 0.3rem;
}

.c0 > div {
  margin: 0.9375rem 0;
  padding-right: 1rem;
  min-width: 13rem;
}

<ol>
  <li>
    <dl
      className="c0"
    >
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Arbeidsgiver
        </dt>
        <dd
          className="typo-normal"
        >
          Johnsen - Kristensen
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Kontonummer
        </dt>
        <dd
          className="typo-normal"
        >
          <span
            className="c1"
          >
            <span>
              8786
            </span>
            <span>
              52
            </span>
            <span>
              75970
            </span>
          </span>
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Inntekstsperiode
        </dt>
        <dd
          className="typo-normal"
        >
          MÃ¥nedssats
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Inntekt for perioden
        </dt>
        <dd
          className="typo-normal"
        >
          39,691.00 NOK
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Refusjonstype
        </dt>
        <dd
          className="typo-normal"
        >
          Ikke refusjon
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Refusjon til dato
        </dt>
        <dd
          className="typo-normal"
        >
          12.03.2018
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Sykepenger fra og med
        </dt>
        <dd
          className="typo-normal"
        >
          18.02.2018
        </dd>
      </div>
    </dl>
  </li>
</ol>
`;
